src/main/java/teammates/ui/controller/StudentCourseJoinAction.java

```diff
@@ -20,12 +22,22 @@
     @Override
     public ActionResult execute() {
         Assumption.assertPostParamNotNull(Const.ParamsNames.REGKEY, regkey);
-        String nextUrl = getNextUrl();
         
         statusToAdmin = "Action Student Clicked Join Link"
                         + (account.googleId == null ? "<br>Email: " + account.email
                                                     : "<br>Google ID: " + account.googleId + "<br>Key: " + regkey);
         
+        if (student == null) {
+            statusToAdmin += "<br>Student course join failed due to being deleted by instructor.";
+            String courseId = getRequestParamValue(Const.ParamsNames.COURSE_ID);
+            Assumption.assertNotNull(courseId);
```

Comment:
Let this one be for now. @Neurrone if you're interested, #5532 #2972 are for you.